Cannot use departmental budget cards as source of credit withdraw

Pepsilawn opened the pull request:
## About The Pull Request

Currently you can put cargo's budget card into your PDA, go into NT Pay and send over the budget to any pay token you want including yours. This just adds a check to ensure you aren't using that kind of card as withdrawal source.

## Why It's Good For The Game

The cargo budget card isn't withdrawable from outside of the loud machine in the most secure station area that yells to everyone in Common comms to come beat you up the moment you use it as evidented by the card's code comments. This fixes a cheesy way to bypass the alt+click withdrawal restriction that was just having an extra step of putting it in your PDA before doing it.

## Changelog
:cl:
fix: you cannot use departmental budget cards as withdrawal sources for NT Pay anymore
/:cl:

Pepsilawn committed:
Cannot use departmental budget cards as source of credit withdraw

Pepsilawn committed:
Merge branch 'master' into departmental-card-on-nt-pay

JohnFulpWillard commented on code/modules/modular_computers/file_system/programs/nt_pay.dm:
```diff
@@ -20,6 +20,9 @@
 /datum/computer_file/program/nt_pay/ui_act(action, list/params, datum/tgui/ui)
 	switch(action)
 		if("Transaction")
+			if(istype(current_user, /datum/bank_account/department))
+				return to_chat(usr, span_notice("The app is unable to withdraw from that card."))
```
Currently we check the type for ``/obj/item/card/id/departmental_budget`` in these cases (like the roulette table).

We really should make a helper for this, maybe like ``#define IS_DEPARTMENTAL_ACCOUNT(card) (card in SSeconomy.dep_cards)``, and replace all instances with that (and preferably, this one too)

Pepsilawn replied:
Yeah thats a good idea wait a sec

Pepsilawn committed:
Replaces istype checks for dep accounts with a new define

Replaces istype checks for dep accounts with a new define
Renames generated_accounts to departmental_accounts

JohnFulpWillard reviewed (approved):
based as hell
